pkg/apis/pipeline/v1beta1/pipeline_validation_test.go

```diff
@@ -916,6 +916,14 @@ func TestValidatePipelineParameterVariables_Failure(t *testing.T) {
 				Name: "a-param", Value: ArrayOrString{Type: ParamTypeArray, ArrayVal: []string{"value: $(params.baz[*])", "last"}},
 			}},
 		}},
+	}, {
+		name: "multiple parameters with the same name",
+		params: []ParamSpec{{
+			Name: "baz", Type: ParamTypeString, Default: &ArrayOrString{Type: ParamTypeString, StringVal: "astring"},
+		}, {
+			Name: "baz", Type: ParamTypeString, Default: &ArrayOrString{Type: ParamTypeString, StringVal: "astring"},
+		}},
+		tasks: []PipelineTask{},
```

Comment:
can we add a single valid task such that this test confirms that the pipeline validation does not fail because of no tasks but fails because of duplicate params?

Reply:
The empty task list is valid though at least in the current state. So there should be no difference between an empty list and one valid task. Considering that, if you still prefer to have a single valid task here let me know and I'll add it.

Reply:
yes please, so that it isolates the tests being done 🙏

Reply:
done

Reply:
thanks for adding more tests, it now has some coverage for the error being added.

Before:

<img width="1101" alt="Screen Shot 2020-06-16 at 9 39 56 AM" src="https://user-images.githubusercontent.com/206285/84802544-9ca19d00-afb5-11ea-804f-4c34250246f8.png">

Now:

<img width="1082" alt="Screen Shot 2020-06-16 at 9 40 22 AM" src="https://user-images.githubusercontent.com/206285/84802573-a3301480-afb5-11ea-969d-ebfdbccf0278.png">